lib/chef/provider/apt_update.rb

```diff
@@ -62,16 +58,17 @@ def apt_up_to_date?
 
       def do_update
         [STAMP_DIR, APT_CONF_DIR].each do |d|
-          build_resource(:directory, d, caller[0]) do
+          declare_resource(:directory, d, caller[0]) do
             recursive true
-          end.run_action(:create)
+          end
         end
 
-        build_resource(:file, "#{APT_CONF_DIR}/15update-stamp", caller[0]) do
+        declare_resource(:file, "#{APT_CONF_DIR}/15update-stamp", caller[0]) do
           content "APT::Update::Post-Invoke-Success {\"touch #{STAMP_DIR}/update-success-stamp 2>/dev/null || true\";};"
-        end.run_action(:create_if_missing)
+          action :create_if_missing
+        end
 
-        shell_out!("apt-get -q update")
+        declare_resource(:execute, "apt-get -q update", caller[0])
```

Comment:
Is there a strong reason to do this instead of using the DSL? Feels weird and excessively verbose.